src/types/decimal.rs

```diff
@@ -296,8 +296,37 @@ impl TryFrom<f64> for Decimal {
 
 impl Display for Decimal {
     fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
-        // TODO: This is correct, but not the most human-friendly format.
-        write!(f, "{}d{}", self.coefficient, self.exponent)
+        let digits = &*self.coefficient.magnitude.to_string();
+        let len = digits.len();
+        // The index of the decimal
+        let i_d = len as i64 + self.exponent;
+
+        if self.coefficient.sign == Sign::Negative {
+            write!(f, "-").unwrap();
+        };
+
+        if self.exponent > 6 {
+            // e.g. A.BCd12
+            write!(f, "{}.{}d{}", &digits[0..1], &digits[1..len], (i_d - 1))
+        } else if self.exponent > 0 {
+            // e.g. ABC0000. or ABC.
+            let zeroes = "0".repeat(self.exponent as usize);
```

Comment:
This line is _adding_ more digits of precision to the output.

Reply:
Good callout. I've fixed it and the tests pass now.